src/adapter/src/coord.rs

```diff
@@ -3138,6 +3138,34 @@ impl<S: Append + 'static> Coordinator<S> {
                 },
             )
             .await;
+
+        let storage_sink_from_entry = self.catalog.get_entry(&sink.from);
+        let storage_sink_desc = mz_storage::types::sinks::SinkDesc {
+            from: sink.from,
+            from_desc: storage_sink_from_entry
+                .desc(&self.catalog.resolve_full_name(
+                    storage_sink_from_entry.name(),
+                    storage_sink_from_entry.conn_id(),
+                ))
+                .unwrap()
+                .into_owned(),
+            connection: SinkConnection::Tail(TailSinkConnection {}),
+            envelope: Some(sink.envelope),
+            as_of: SinkAsOf {
+                frontier: Antichain::new(),
+                strict: false,
+            },
+        };
+
+        // TODO(chae): This is where we'll create the export/sink in storaged
+        let _ = self.controller.storage_mut().create_exports(vec![(
+            id,
+            ExportDescription {
+                sink: storage_sink_desc,
+                remote_addr: None,
+            },
+        )]);
```

Comment:
This bit doesn't check out for me! L3143-3158 looks duplicated from L3118-3135—you could just return the desc that you need from that closure. That said, this whole bit is just a hack to validate that the catalog will allow the sink to be created with a dummy tail sink before we actually create the Kafka topic. 

ISTM the place you want to be doing this is in handle_sink_connection_ready?